### PR TITLE
[PLT-677] Basic gRPC mock generator

### PR DESCRIPTION
**Pull request by rogchap**
This is the `protoc` plugin to generate mock data based on gRPC service methods.
See README for features currently implemented.

### CODE REVIEW

File: protobuf/protoc-gen-gogrpcmock/plugin/plugin.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
+// Copyright (c) 2019 SafetyCulture Pty Ltd. All Rights Reserved.
```
Comment (SafetyTravis) [inline protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: This file is not following the recommended project structure

Comment (SafetyTravis) [inline protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: This file is not following the recommended project structure

Comment (SafetyTravis) [inline protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: This file is not following the recommended project structure

Comment (SafetyTravis) [inline protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: This file is not following the recommended project structure

Comment (SafetyTravis) [inline protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: This file is not following the recommended project structure

File: protobuf/protoc-gen-gogrpcmock/plugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	g.In()
+
+	for _, field := range msg.Field {
+		fieldName := g.GetFieldName(msg, field)
+		fieldType, _ := g.GoType(msg, field)
+		repeated := field.IsRepeated()
+		nullable := gogoproto.IsNullable(field) && field.IsMessage()
+
+		if field.OneofIndex != nil {
+			// TODO: Deal with oneof
```
Comment (pawanrawal) [inline protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: Should we be adding an INFO log here and at other places to say we are skipping some fields?

Comment (rogchap) [reply protobuf/protoc-gen-gogrpcmock/plugin/plugin.go]: Will be addressing these soon; I just wanted to get a first pass working.

**Pull request merged**